### PR TITLE
Avoid inlining of arrow kernel dispatcher

### PR DESCRIPTION
**Pull request by vitstn**
### Changelog entry 

...

### Changelog category 

* Not for changelog (changelog entry is not required)

### Additional information

...

### CODE REVIEW

File: ydb/library/yql/minikql/invoke_builtins/mkql_builtins_string_kernels.cpp [hunk 1]
```diff
@@ -12,149 +12,265 @@ constexpr auto GetArgumentsCount(Return(*)(Args...)) noexcept
...
-        const char* data1 = arr1.GetValues<char>(2, 0);
-        const char* data2 = arr2.GetValues<char>(2, 0);
-        for (int64_t i = 0; i < length; ++i) {
-            std::string_view val1(data1 + offsets1[i], offsets1[i + 1] - offsets1[i]);
-            std::string_view val2(data2 + offsets2[i], offsets2[i + 1] - offsets2[i]);
-            resValues[i] = TOp::Do(val1, val2);
+    static void ArrayArrayCore(const TOffset1* stringOffsets1, const char* data1,
+        const TOffset2* stringOffsets2, const char* data2, TOutput* resPtr, int64_t length, int64_t offset1, int64_t offset2) {
+        stringOffsets1 += offset1;
+        stringOffsets2 += offset2;
```
Comment (igormunkin) [inline ydb/library/yql/minikql/invoke_builtins/mkql_builtins_string_kernels.cpp:177]: Ditto.

File: ydb/library/yql/minikql/invoke_builtins/mkql_builtins_string_kernels.cpp [hunk 2]
```diff
@@ -12,149 +12,265 @@ constexpr auto GetArgumentsCount(Return(*)(Args...)) noexcept
...
-                    const TOffset1* offsets1 = arr1.GetValues<TOffset1>(1);
-                    for (int64_t i = 0; i < length; ++i) {
-                        resValues[i] = TOp::DoWithEmptyRight(offsets1[i + 1] - offsets1[i]);
-                    }
-                }
+    static void ArrayScalarCore(const TOffset1* stringOffsets1, const char* data1,
+        const TOffset2* stringOffsets2, const char* data2, TOutput* resPtr, int64_t length, int64_t offset1, int64_t offset2) {
+        Y_UNUSED(offset2);
+        const auto val2 = std::string_view(data2, *(const size_t*)stringOffsets2);
+        stringOffsets1 += offset1;
```
Comment (igormunkin) [inline ydb/library/yql/minikql/invoke_builtins/mkql_builtins_string_kernels.cpp:157]: Ditto.

File: ydb/library/yql/minikql/invoke_builtins/mkql_builtins_string_kernels.cpp [hunk 3]
```diff
@@ -12,149 +12,265 @@ constexpr auto GetArgumentsCount(Return(*)(Args...)) noexcept
...
-                    const TOffset2* offsets2 = arr2.GetValues<TOffset2>(1);
-                    for (int64_t i = 0; i < length; ++i) {
-                        resValues[i] = TOp::DoWithEmptyLeft(offsets2[i + 1] - offsets2[i]);
-                    }
-                }
+    static void ScalarArrayCore(const TOffset1* stringOffsets1, const char* data1,
+        const TOffset2* stringOffsets2, const char* data2, TOutput* resPtr, int64_t length, int64_t offset1, int64_t offset2) {
+        Y_UNUSED(offset1);
+        const auto val1 = std::string_view(data1, *(const size_t*)stringOffsets1);
+        stringOffsets2 += offset2;
```
Comment (igormunkin) [inline ydb/library/yql/minikql/invoke_builtins/mkql_builtins_string_kernels.cpp:136]: Ditto.

**Review by igormunkin — CHANGES_REQUESTED** [review]
@vitstn, thanks for the patch! LGTM, but please consider my nits below.

Feel free to ignore the comments, if these are not applicable; `request changes` is set to prevent auto-merge.

File: ydb/library/yql/minikql/invoke_builtins/mkql_builtins_string_kernels.cpp [hunk 4]
```diff
@@ -12,149 +12,265 @@ constexpr auto GetArgumentsCount(Return(*)(Args...)) noexcept
...
+    TUntypedStringUnaryArrayFuncPtr func) {
+
+    const auto& arg = batch.values[0];
+    auto& resArr = *res->array();
+
+    const auto& arr = *arg.array();
+    const auto length = arr.length;
+    const auto resValues = resArr.buffers[1]->mutable_data();
+
+    const auto offsets = arr.buffers[1]->data();
```
[the remark this answers is not in the document]
Comment (vitstn) [reply ydb/library/yql/minikql/invoke_builtins/mkql_builtins_string_kernels.cpp:229]: TOffset type is not available here. So we use generic .data() accessor, which returns const void *

**Review by igormunkin — APPROVED** [review]
Discussed offline, the nits above are inapplicable. LGTM.

**Pull request merged**